### ISSUE TITLE
Typescript generics and await don't work together

### ISSUE THREAD

**Issue by Starkteetje** [semgrep/semgrep#6405]
**Describe the bug**
Matching a function call in typescript that has a generic and is awaited doesn't work.

**To Reproduce**
https://semgrep.dev/s/Zz0x

**Expected behavior**
```
await callService<Asd>(arg);
callService<Asd>(arg);
await callService(arg);
```
all match `callService(...)`

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Environment**
semgrep.dev

**Use case**
Matching function calls with generics in asynchronous functions

**Comment by joshbouncesecurity** [comment]
Yep, I came across the same issue. Super frustrating when I am trying to write custom rules for clients :(
